File: backend/app/schemas/paper.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PaperResponse(BaseModel):
    id: str
    title: str
    summary: str
    authors: list[str]
    categories: list[str]
    published_at: datetime
    updated_at: datetime
    arxiv_url: str
    pdf_url: str | None

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/src/types/paper.ts
export type Paper = {
  id: string;
  title: string;
  summary: string;
  authors: string[];
  categories: string[];
  published_at: string;
  updated_at: string;
  arxiv_url: string;
  pdf_url: string | null;
};

export type PapersResponse = {
  query: string;
  count: number;
  papers: Paper[];
};

export interface PaperAIAnalysis {
  id: number;
  executive_summary: string;
  why_it_matters: string;
  difficulty: "Beginner" | "Intermediate" | "Advanced" | "Research";
  reading_time_minutes: number;
  key_contributions: string[];
  limitations: string[];
  use_cases: string[];
  prerequisites: string[];
  target_roles: string[];
  model_name: string;
  created_at: string;
  updated_at: string;
}

export interface PaperDetailResponse {
  paper: Paper;
  analysis: PaperAIAnalysis | null;
}